ignite/metrics/rouge.py

```diff
@@ -0,0 +1,437 @@
+    - `y` must be a list of sequence of tokens.
+
+    Args:
+        metrics: names of metrics computed. Valid names are "Rouge-L" and "Rouge-n" where n is an integer 1 <= n <= 9.
+        multiref: reduces scores for multi references. Valid values are "best" and "average" (default: "average").
+        alpha: controls the importance between recall and precision (alpha -> 0: recall is more important, alpha -> 1:
+           precision is more important)
+        output_transform (callable, optional): a callable that is used to transform the
+           :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+           form expected by the metric. This can be useful if, for example, you have a multi-output model and
+           you want to compute the metric with respect to one of the outputs.
+        device (Union[str, torch.device]) – specifies which device updates are accumulated on. Setting the metric’s
+           device to be the same as your update arguments ensures the update method is non-blocking. By default, CPU.
+
+    Example:
+
+    .. code-block:: python
+
+        from ignite.metrics import Rouge
+
+        m = Rouge(metrics=["Rouge-L", "Rouge-2"], multiref="best")
+
+        candidate = "the cat is not there".split()
+        references = [
+            "the cat is on the mat".split(),
+            "there is a cat on the mat".split()
+        ]
+
+        m.update((candidate, references))
+
+        m.compute()
+        >>> {'Rouge-L-P': 0.6, 'Rouge-L-R': 0.5, 'Rouge-L-F': 0.5, 'Rouge-2-P': 0.5, 'Rouge-2-R': 0.4, 'Rouge-2-F': 0.4}
+
+        .. versionadded:: 0.5.0
+        """
+
+    def __init__(
+        self,
+        metrics: Sequence[str] = None,
+        multiref: str = "average",
+        alpha: float = 0,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ):
+        if len(metrics) == 0:
+            raise ValueError("Rouge must have at least one metric")
+        self.internal_metrics = []
+        for m in metrics:
+            if m == "Rouge-L":
+                self.internal_metrics.append(RougeL(multiref, alpha, output_transform, device))
+            else:
+                match = re.match("Rouge-(?P<ngram>[1-9])", m)
+                if not match:
+                    raise ValueError("Rouge metric name must be 'Rouge-L' or 'Rouge-N' for N in [1 ... 9]")
+                n = int(match.group("ngram"))
+                self.internal_metrics.append(RougeN(n, multiref, alpha, output_transform, device))
+        super(Rouge, self).__init__(output_transform=output_transform, device=device)
+
+    @reinit__is_reduced
+    def reset(self) -> None:
+        for m in self.internal_metrics:
+            m.reset()
+
+    @reinit__is_reduced
+    def update(self, output: Tuple[Sequence[Any], Sequence[Sequence[Any]]]) -> None:
+        for m in self.internal_metrics:
+            m.update(output)
+
+    @sync_all_reduce()
```

Comment:
I think we do not need to decorate the method if there is nothing to reduce ...

Reply:
If the method is not decorated, a warning is raised when run in parallel. However, `Rouge` is a simple proxy to `RougeL` and `RougeN` which are parallel. If you have an idea, I take it !

Reply:
What kind of warning it is ? I'm thinking about `MetricsLambda.compute` which is not decorated:
https://github.com/pytorch/ignite/blob/cc76de461f63475f3b792c7c109fede95301556e/ignite/metrics/metrics_lambda.py#L88-L92

Reply:
The warning below 

https://github.com/pytorch/ignite/blob/cc76de461f63475f3b792c7c109fede95301556e/ignite/metrics/metric.py#L215